content/articles/how-to-implement-structured-logging-in-adonisjs/index.md

```diff
@@ -0,0 +1,263 @@
+
+To install AdonisJS 5, run the command below:
+
+```bash
+npm init adonis-ts-app@latest forum-api
+```
+
+The command creates a new Adonis 5 project, properly configured and ready to be used. We will continue with the Forum API developed with the article on [How to Implement Caching using Adonis.js 5](/engineering-education/how-to-implement-caching-in-adonisjs-5/). You can [clone the repository](https://github.com/Kaperskyguru/adonisjs-forum-api) for an easy setup.
+
+### Setting up Adonis Logger
+
+AdonisJS ships with an [inbuilt logger](https://docs.adonisjs.com/guides/logger) developed on top of one of the fastest logging libraries in Node.js called [Pino](https://github.com/pinojs/pino).
+
+Log messages can cause a throttling effect on the application due to the overtime addition of log messages. Such an effect includes reduced requests per second.
+
+[Pino](https://github.com/pinojs/pino) is a low overhead Node.js logger using minimum resources for Logging and is 5X faster than other alternatives. See the [Benchmarks](https://getpino.io/#/docs/benchmarks) document for comparisons.
+
+You can import and use the logger directly as follows without having to install any package:
+
+```js
+import Logger from "@ioc:Adonis/Core/Logger";
+
+Logger.info("A info message");
+Logger.warn("A warning");
+```
+
+### Handling Logging in Adonis 5
+
+You should have cloned the Forum API Project and correctly test and configure it to work correctly.
+
+We will be using AdonisJS default Logger for all the log operations. Let’s look at how to configure the logger.
+
+Firstly, open `config/app.ts`, and change the value of `generateRequestId` to `true` , allowing Adonis to generate a unique Request ID to identify each request coming to the AdonisJS server.
+
+```js
+//...
+
+/*
+    |--------------------------------------------------------------------------
+    | Request Ids
+    |--------------------------------------------------------------------------
+    |
+    | Setting this value to `true` will generate a unique request id for each
+    | HTTP request and set it as `x-request-id` header.
+    |
+    */
+    generateRequestId: true,
+
+// ...
+```
+
+If your application hits any errors along the way, it will generate a unique request ID for that request and log in to the console for further investigations.
+
+![Log Errors](/engineering-education/how-to-implement-structured-logging-in-adonisjs/log_errors.png)
+
+Next, we will update our Forum API to log out specific and vital information of our application as the user interacts with it in production.
+
+First, we will update the `ForumsController.ts` in `app/Controllers/Http`:
+
+```js
+import { HttpContextContract } from "@ioc:Adonis/Core/HttpContext";
+import Forum from "App/Models/Forum";
+import Logger from "@ioc:Adonis/Core/Logger";
```

Comment:
For all the code blocks:
- Try keeping them as short as possible. 
- Don't add unused code. No repetition.
- Break it up into chunks for the reader to follow along. 
- Explain what this file does, why we are changing it, and what to expect after applying the changes.
- Don't put all the changes for a single file in a single block. Changes in different locations in the file should be discussed individually, with clear explanation and instructions.
- All of the above should be in a paragraph just above the code block.

Please apply the above to all your code blocks.